src/checkers/inference/InferenceTreeAnnotator.java

```diff
@@ -276,6 +293,25 @@ public Void visitNewClass(final NewClassTree newClassTree, final AnnotatedTypeMi
 
         annotateMethodTypeArgs(newClassTree);
 
+
+        if (newClassTree.getClassBody() != null) {
+            // For a fully annotated anonymous class instantiation as follows,
+            //     new @VarAnnot(1) A() @VarAnnot(2) {...}
+            // create the implied equality constraint "1 == 2"
+            ConstraintManager constraintManager = InferenceMain.getInstance().getConstraintManager();
+            SlotManager slotManager = InferenceMain.getInstance().getSlotManager();
+
+            // Get the varSlot on the type identifier
+            final Slot identifierSlot = slotManager.getSlot(atm);
+            AnnotatedTypeMirror classType = atypeFactory.getAnnotatedType(newClassTree.getClassBody());
+            // Get the varSlot on the anonymous class body
+            final Slot classBodySlot = slotManager.getSlot(classType);
+            if (classBodySlot instanceof SourceVariableSlot) {
```

Comment:
How could `classBodySlot` ever not be a `SourceVariableSlot`? The location for 2 is not syntactically valid, so must have been created before. Can you raise an error to make sure it doesn't happen, or just do the cast and let that fail.